### ISSUE TITLE
discontinued? redeclarations errors on compile

### ISSUE THREAD

**Issue by mckaygerhard** [arcadenea/3doh#1]
i have some interest on only cmpile and runs, for my olders computers, i known that need power, but i make run a n64 emulator in a p2 machine at 256Mb ram, so 3do can be possible to run emulated in a P4 machine like!

i clone and try to compile, after some changes (due u mixed without conditioned the opengl and raster only compilations) i got that error:

```
video.c:147: error: using typedef-name ‘BitmapCrop’ after ‘struct’
./freedo/freedocore.h:58: error: ‘BitmapCrop’ has a previous declaration here
video.c:147: error: invalid type in declaration before ‘;’ token
video.c:154: error: invalid conversion from ‘int*’ to ‘int’
./freedo/frame.h:26: error: too many arguments to function ‘void Get_Frame_Bitmap(VDLFrame*, void*, int, int, int, int, int, int)’
video.c:154: error: at this point in file
video.c:158: error: ‘loadTexture’ was not declared in this scope
video.c:158: error: ‘videoDrawGL’ was not declared in this scope
video.c: In function ‘GLuint loadTexture(SDL_Surface*)’:
make: *** [video.o] Error 1
```

i dont care about moder compilators, i used older systems and olders versions of gcc (<< 4.4), older opengl and sdl 1.2 rather than 2

of course i dont use any winbuntu like OS, geento and debian if this cares!

**Comment by arcadenea** [comment]
Well, it must work with older systems and older versions of gcc. I've tested 3doh on an ARM powered tablet with debian (allwinner a10) and games like wolfestein runs fine.
3doh it's not discontinued, i just don't have much time now for development, but let me check out if i can fix the errors you're getting.

**Comment by mckaygerhard** [comment]
i'm using gcc 4.1, and 4.4, also using sdl 1.2.14 (with the patch for x11 driver of 1.2.15), the makefiles need some checks (but dont care of i can doit by myself), for opengl vs softwonly builds

umm i visit u'r page.. very interesting info , specially due are for linux! i can complete more u'r blog info's but better make it a page in github.. i'll post inspired on u a wiki emulation for linux based on debian systems here: https://github.com/mckayemu/emulation/wiki

**Comment by arcadenea** [comment]
i remember that 3doh even compiles against the Dreamcast toolchain (i think i've used gcc 4.4 in the time i've tested). Of course, it runs at 1 fps hehehe
Emulation on linux is poorly documented and a lot of emus are outdated, some things are changing since Raspberry Pi is on the market...seems like people are getting again in the groove. I'm building an arcade machine with one old ARM tablet (1 ghz single core cpu / 512 mb of ram), and a lot of linux emus works great: gngeo, mame, 3doh ;)...i want to port/improve some others like franxis' Mame4all, Chui's Neo4All and maybe a cps2 emu (fba maybe?)

**Comment by mckaygerhard** [comment]
the XXXX4all have th same problems, then due that concluded that 3doh never was tested for x86.. only for arm.. maybe that's the problem (and with that port the xxx4all emulators will be a pain of time) so for porting those emulators a good start (je jeje) must be fixed the errors cited by me!

about the documentation on linux emulators, dont worry let me that to me! u will see... i have a good amount of resources, my brother have the most huge amount of roms and compatibility checks done, i made a huge amount of checks and tests, and currently i played with a 7 yeasr old boy the final fantasy V in spanish so was patched with ucon64 suite for linux (a editin tool for roms 8<->32 bits)

i'll post later a recent menu of lxde emulation suite, wheantime here a menu using kde in sage from ol days ![kde-emulation-sarge](http://2.bp.blogspot.com/_WfPssHN5G0c/ScVt2TtZk8I/AAAAAAAAAQU/fVY02IwQQ1Y/s1600/eulators.png)

**Comment by arcadenea** [comment]
where are you from? seems like you are spanish speaker (i'm from Argentina). 3doh was tested in x86 and x64 architectures, of course, it's not easy to develop a multiplatform emu, and i know that my code is awful in a lot of places. Let me see if i can reproduce the errors that you're getting...

**Comment by mckaygerhard** [comment]
yeah spanish.. but my bad english its due i'm a "hurry writter" sorry for that, i not have 24/7 internet connection

i'll waith u'r changes , i'll be wacht this repo for...
